tests/src/Interop/SymVer/SymVerManaged.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace PInvokeTests
+{
+    class SymVerTests
+    {
+        #region PInvoke declarations
+
+        [DllImport("SymVerNative", EntryPoint="foo")]
+        private static extern int foo_10();
+
+        [DllImport("SymVerNative", EntryPoint="foo@VERS_1.1")]
+        private static extern int foo_11();
+
+        [DllImport("SymVerNative", EntryPoint="foo@VERS_1.2")]
+        private static extern int foo_12();
+
+        #endregion
+
+        #region test methods
+
+        public static void TestSupported()
+        {
+            try
+            {
+                Assert(foo_10() == 10, "Call to foo_10 should return 10");
+                Assert(foo_11() == 11, "Call to foo_11 should return 11");
+                Assert(foo_12() == 12, "Call to foo_12 should return 12");
+            }
+            catch (Exception e)
+            {
+                Assert(false, e.ToString());
+            }
+        }
+
+        public static void TestUnsupported()
+        {
+            bool exceptionThrown = false;
+            try
+            {
+                foo_11();
+            }
+            catch (Exception)
+            {
+                exceptionThrown = true;
+            }
+            Assert(exceptionThrown, "Call to foo_11 should throw an exception");
+            // TODO: check for more specific exception
```

Comment:
Probably this is throwing EntryPointNotFoundException now, but PlatformNotSupportedException may be a better fit.